[stable10] Move excluded files to config array

sharidas opened the pull request:
Moving excluded files to config array.
Instead of populating in the ExcludeFile*
php files, we can have it in config.php.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please fill out below information carefully.

Please note that any kind of change first has to be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.
-->

## Description
<!--- Describe your changes in detail -->
This change would help users to mention folders/files to be excluded during integrity check to the config.php. In the config.php it will be represented as an array.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/owncloud/core/issues/23186
https://github.com/owncloud/enterprise/issues/1874

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Instead of hard coding the files/folders in the files, its better to move them to the config.php. Also this change would help the integrity check to exclude files/folders mentioned in the config.php

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

sharidas committed:
[stable10] Move excluded files to config array

Moving excluded files to config array.
Instead of populating in the ExcludeFile*
php files, we can have it in config.php.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

PVince81 reviewed (approved):
:+1: if you updated this with the latest changes from the master PR (PHPDoc changes I believe)

phil-davis commented:
This is a backport of #29126

sharidas commented:
@PVince81 Yes I have updated this PR from the master branch. Replaced `mixed` with `array` in the PHPDoc - https://github.com/owncloud/core/pull/29460/files#diff-8c88c803f1798e69a75cd73cbaba2a12R233

bcutter commented:
Hell I´m searching for the syntax in config.php for 20 minutes because the [mentioned](https://doc.owncloud.org/server/10.0/admin_manual/release_notes.html) config.sample.php does NOT contain any information regarding this... pretty frustrating. Is it my fault or is it just poor documentation?

PVince81 commented:
@bcutter it's called "integrity.excluded.files", you can see it directly in config.sample.php.

I see that the documentation hasn't been synced with the contents of config.sample.php, raised https://github.com/owncloud/documentation/issues/3638.

Thanks for spotting.

bcutter commented:
Yes the name is known, but what‘s the syntax vor keys? And values absolute or relative paths? Maybe you can just link the config.sample.php here.

PVince81 commented:
https://github.com/owncloud/core/blob/v10.0.4/config/config.sample.php#L1246

It's relative paths

settermjd commented on lib/private/IntegrityCheck/Checker.php:
```diff
@@ -382,13 +382,20 @@ private function verify($signaturePath, $basePath, $certificateCN, $force = fals
 			throw new InvalidSignatureException('Signature could not get verified.');
 		}
 
+		//Exclude files which shouldn't fall for comparison
```
In this line, it seems "fall" is incorrect or unclear. @sharidas, can you clarify, please?

bcutter commented:
Hmm, why it´s not working?

**config.php:**
```
  'integrity.excluded.files' =>
  array (
    '.htaccess_OLD',
        ),
```

**Integrity check after rescan is still:**
```
	- EXTRA_FILE
		- .htaccess_OLD
```
